### DISCUSSION TITLE
Xcode github ssh key no longer works

### DISCUSSION THREAD

**Discussion by marianneji** [community/community#23700] [New to GitHub]
<p>Hi Guys,</p>
<p>I commit and push via Xcode to my github account everyday with a ssh key and it worked just fine till this morning! I had a mail by github, yesterday, username and password is deprecated please refer to developer guide.<br>
I have generated a new key via terminal, and assigned it to ssh agent, register on xcode… still doesn’t work!<br>
as a response from Xcode :<br>
failed to connect to <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>: Connection refused (-1)<br>
as a response from terminal:</p>
<p>ssh: connect to host <a href="http://github.com" rel="noopener nofollow ugc">github.com</a> port 22: Connection refused</p>
<p>fatal: Could not read from remote repository.</p>
<p>Please make sure you have the correct access rights</p>
<p>and the repository exists.</p>
<p>I don’t see how I can do ! Can someone help me?</p>
<p>Thanks a lot guys</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey <a class="mention" href="/marianneji">@marianneji</a> :wave:</p>
<p>How are you going? Thanks so much for your post!</p>
<p>The <code>Connection refused</code>  message typically suggests that port 22 is being blocked on your network, somehow.</p>
<p>Though I would start by verifying that the key you’re using is still valid (not expired) and has permission to connect to the repository you’re trying to work with. You can also use this utility:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <a href="https://github-debug.com/" target="_blank" rel="noopener">github-debug.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://github-debug.com/" target="_blank" rel="noopener">GitHub Debug</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>…to help test your connection.</p>
<p>If you see anything confusing in your debug tests, please feel free to post-back. However, be very mindful of what you choose to include, and please obscure any sensitive information that might be exposed.</p>
<p>Let us know how you do!</p>

**Comment by marianneji** [comment]
<p>Hey <a class="mention" href="/nethgato">@nethgato</a>,</p>
<p>Thank you so much for your response. You were right, I’m using my laptop at my new work and the network is restricted… That’s why I couldn’t commit and push my repo. I have generated a new Key, and I use my Iphone to push on github with the terminal.</p>
<p>Thanks again, that was really helpful :wink:</p>
